From 0b1495c63a885f8dd3c72c053b5ac228caa85b16 Mon Sep 17 00:00:00 2001
From: Rohith Sharma K S <rohithsharmaks@apache.org>
Date: Thu, 24 Nov 2016 12:18:38 +0530
Subject: [PATCH 2171/2848] YARN-5920. Fix deadlock in
 TestRMHA.testTransitionedToStandbyShouldNotHang.
 Contributed by Varun Saxena.

(cherry picked from commit e15c20edba1e9a23475ee6a4dfbadbdb8c1f668a)

Change-Id: I7b951f5c984d945dd15a4b4f75f4685e32e9f378
---
 .../yarn/server/resourcemanager/TestRMHA.java      |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestRMHA.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestRMHA.java
index 5eb3025..2d59863 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestRMHA.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestRMHA.java
@@ -463,8 +463,7 @@ public void testTransitionedToStandbyShouldNotHang() throws Exception {
 
     MemoryRMStateStore memStore = new MemoryRMStateStore() {
       @Override
-      public synchronized void updateApplicationState(
-          ApplicationStateData appState) {
+      public void updateApplicationState(ApplicationStateData appState) {
         notifyStoreOperationFailed(new StoreFencedException());
       }
     };
-- 
1.7.9.5

